community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java

```diff
@@ -119,6 +119,19 @@ public CommonAbstractStore(
         }
         catch ( Exception e )
```

Comment:
We recently had to face a similar situation where we needed to close resources on exception. The problem there was that some of the exceptions caught actually caused the close to hang and we never got to see the error in any of the logs. The solution was obviously to log first, then try to cleanup (and then throw). I wonder if we should do the same thing here.